Source/backend/models/Cart.js

```diff
@@ -0,0 +1,21 @@
+import mongoose from 'mongoose';
+
+const cartSchema = new mongoose.Schema({
+  products: [{
+    productId: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: 'Product',
+      required: true,
+    },
+    quantity: {
+      type: Number,
+      required: true,
+    },
+  }],
+  totalPrice: {
+    type: Number,
+    default: 0,
+  },
+});
+
```

Comment:
It’s good to add code that automatically calculates the totalPrice based on products and their quantities. This keeps the data consistent